spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/services/authService/SCServerAuthMethod.java

```diff
@@ -0,0 +1,139 @@
+    private static final String JSON_WEB_TOKEN = "x-access-token";
+    private static final String USERNAME = "username";
+    private static final String PWD = "pwd";
+    private Context context;
+    private String jsonWebToken;
+    private SecureSharedPreferences settings;
+    private String serverUrl;
+
+
+    public SCServerAuthMethod(Context context, String serverUrl ) {
+        this.context = context;
+        this.settings = new SecureSharedPreferences(context);
+        this.serverUrl = serverUrl;
+    }
+
+    @Override
+    public boolean authFromCache() {
+        String u = getUsername();
+        String p = getPassword();
+        if (u != null && p != null) {
+            return authenticate(u,p);
+        } else {
+            return false;
+        }
+    }
+
+    @Override
+    public boolean authenticate(String username, String pwd) {
+        return auth(username, pwd);
+    }
+
+    @Override
+    public void logout() {
+        removeCredentials();
+    }
+
+    @Override
+    public String xAccessToken() {
+        return settings.getString(JSON_WEB_TOKEN, null);
+    }
+
+    @Override
+    public String username() {
+        return getUsername();
+    }
+
+    private boolean auth(final String username, final String pwd) {
+        boolean authed = false;
+        final String theUrl = String.format(Locale.US, "%s/api/authenticate", serverUrl);
+        Response response = HttpHandler.getInstance()
+                .post(theUrl, String.format("{\"email\": \"%s\", \"password\":\"%s\"}", username, pwd))
+                .toBlocking()
+                .first();
+
+        if (response.isSuccessful()) {
+            try {
+                jsonWebToken = new JSONObject(response.body().string())
+                        .getJSONObject("result").getString("token");
+                if (jsonWebToken != null) {
+                    saveAccessToken(jsonWebToken);
+                    saveCredentials(username, pwd);
+                    authed = true;
+                }
+            } catch (JSONException e) {
+                e.printStackTrace();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        } else {
+            authed = false;
+            logout();
+        }
+        return authed;
+    }
+
+    private void saveAccessToken(String accessToken) {
+        SecureSharedPreferences.Editor editor = settings.edit();
+        editor.putString(JSON_WEB_TOKEN, accessToken);
+        editor.commit();
+    }
+
+    private void saveCredentials(String username, String password) {
+        SecureSharedPreferences.Editor editor = settings.edit();
+        editor.putString(USERNAME, username);
+        editor.putString(PWD, password);
+        editor.commit();
+    }
+
+    private String getPassword() {
+        SecureSharedPreferences settings = new SecureSharedPreferences(context);
+        return settings.getString(PWD, null);
+    }
+
+    private String getUsername() {
```

Comment:
Maybe we should get rid of this method and just use `username()` b/c that's what's required by the interface?